File: JavaScript/1st_ex/main.js
function err(exc) {
    throw new Error(exc);
}

function init() {
    let val = parseInt(prompt(), 10);
    if (isNaN(val) || val <= 0) {
        err("Invalid input");
    }
    return val;
}

function func(S, N, M) {
    if (S * M > (S - Math.floor(S / 7)) * N) {
        return -1;
    }
    return Math.ceil(S * M / N);
}

try {
    let S = init();
    let N = init();
    let M = init();

    console.log(func(S, N, M));
} catch (e) {
    console.error(e.message);
}

File: c++/1st_ex/main.cpp
#include <cmath>
#include <iostream>

using namespace std;

int init(int val);
int func(int S, int M, int N);

int main() {

  int S, M, N;

  S = init(S);
  N = init(N);
  M = init(M);

  cout << func(S, M, N) << endl;
  return 0;
}

void err(string promt) {
  cerr << promt << endl;
  exit(EXIT_FAILURE);
}

int init(int val) {
  if (!(cin >> val) || val <= 0)
    err("Invalid input");

  return val;
}

int func(int S, int M, int N) {
  if (S * M > (S - S / 7) * N) {
    return -1;
  }
  return ceil((double)S * M / N);
}
